scripts/mod_loader/modui/palette_arrange.lua

```diff
@@ -0,0 +1,355 @@
+	local modcontent = modApi:getCurrentModcontentPath()
+
+	currentPaletteOrder = {}
+	for _, button in ipairs(content.children) do
+		if button.paletteId ~= nil then
+			table.insert(currentPaletteOrder, button.paletteId)
+		end
+	end
+
+	local new_paletteOrder = {}
+	for i = 2, 11 do
+		new_paletteOrder[i] = currentPaletteOrder[i]
+	end
+
+	sdlext.config(modcontent, function(obj)
+		obj.paletteOrder = new_paletteOrder
+	end)
+end
+
+local function buildSdlColorMapBase()
+	colorMapBase = {}
+
+	local basePalette = GetColorMap(1)
+	for i, gl_color in ipairs(basePalette) do
+		local rgb = sdl.rgb(gl_color.r, gl_color.g, gl_color.b)
+		colorMapBase[i*2-1] = rgb
+		colorMapBase[i*2] = rgb
+	end
+
+	return colorMapBase
+end
+
+local function buildSdlColorMap(palette)
+	local res = shallow_copy(colorMapBase)
+
+	for i = 1, 8 do
+		local gl_color = palette[i]
+		if gl_color ~= nil then
+			res[i*2] = sdl.rgb(gl_color.r, gl_color.g, gl_color.b)
+		end
+	end
+
+	return res
+end
+
+local function buildPaletteName(id)
+	return GetText("Palette_Name_"..id:gsub("%s", "_"))
+end
+
+local new_rect = sdl.rect(0,0,0,0)
+local function buildContractedDeco(marginx, marginy, uiDeco, ...)
+	local deco = uiDeco(...)
+	deco.marginx = marginx
+	deco.marginy = marginy
+
+	deco.draw = function(self, screen, widget)
+		local old_rect = widget.rect
+
+		new_rect.x = old_rect.x + self.marginx
+		new_rect.y = old_rect.y + self.marginy
+		new_rect.w = old_rect.w - self.marginx * 2
+		new_rect.h = old_rect.h - self.marginy * 2
+
+		widget.rect = new_rect
+		uiDeco.draw(self, screen, widget)
+		widget.rect = old_rect
+	end
+
+	return deco
+end
+
+local function uiSetDraggable(ui)
+	ui:registerDragMove()
+
+	ui.startDrag = function(self, mx, my, btn)
+		UiDraggable.startDrag(self, mx, my, btn)
+		if self.parent ~= content then return end
+
+		local draggableIndex = list_indexof(content.children, self)
+		self:detach()
+		self:addTo(scrollarea, 1)
+
+		table.remove(content.children, list_indexof(content.children, placeholder))
+		table.insert(content.children, draggableIndex, placeholder)
+
+		placeholder:show()
+		content:relayout()
+	end
+
+	ui.stopDrag = function(self, mx, my, btn)
+		UiDraggable.stopDrag(self, mx, my, btn)
+		if self.parent ~= scrollarea then return end
+
+		local placeholderIndex = list_indexof(content.children, placeholder)
+		table.remove(content.children, placeholderIndex)
+
+		self:detach()
+		self:addTo(content, placeholderIndex)
+
+		table.insert(content.children, placeholder)
+
+		placeholder:hide()
+	end
+
+	ui.dragMove = function(self, mx, my)
+		UiDraggable.dragMove(self, mx, my)
+		if self.parent ~= scrollarea then return end
+
+		for i = 2, #content.children do
+			local ui = content.children[i]
+			if rect_contains(ui.rect, mx, my) then
+				if ui ~= placeholder then
+					table.remove(content.children, list_indexof(content.children, placeholder))
+					table.insert(content.children, i, placeholder)
+				end
+
+				if #content.children > 11 then
```

Comment:
Since the constant `11` appears several times throughout, it might make sense to put it in a variable, similar to how pilot arrange screen defines `MAX_PILOTS`

Reply:
I went with file-locals. The constants will likely not have to change unless ITB gets an expansion, in which case everything will have to be reviewed anyways.